redis/src/client.rs

```diff
@@ -67,19 +70,24 @@ impl Client {
 }
 
 /// Options for creation of async connection
+#[cfg(feature = "aio")]
+#[derive(Clone)]
 pub struct AsyncConnectionConfig {
     /// Maximum time to wait for a response from the server
-    response_timeout: Option<std::time::Duration>,
+    pub(crate) response_timeout: Option<std::time::Duration>,
     /// Maximum time to wait for a connection to be established
-    connection_timeout: Option<std::time::Duration>,
+    pub(crate) connection_timeout: Option<std::time::Duration>,
+    pub(crate) shared_sender: SharedSender,
 }
 
+#[cfg(feature = "aio")]
 impl AsyncConnectionConfig {
     /// Creates a new instance of the options with nothing set
     pub fn new() -> Self {
         Self {
             response_timeout: None,
             connection_timeout: None,
+            shared_sender: SharedSender::default(),
```

Comment:
huh. I forget my own words 🤦

Reply:
if we allow passing senders through the configuration, should we still allow setting senders on the connection after it was created? It will simplify the design if not.

Reply:
pushed a commit showing this.

Reply:
@altanozlu WDYT about removing `set_push_sender` entirely in favor of one-time setting in config?

Reply:
Only in mpx and cm right ?

Reply:
yes. I think that the sync client push behavior needs a bit more refactoring - ATM it's not really usable, and since we've dropped the push manager, we can use different types of senders between the sync & async clients, so I'll probably either remove the push sender entirely, or replace it with a sender from std instead of tokio, to remove the tokio dependency from non-async builds.

Reply:
Why you think sync client's push behavior is not usable, because of not having new data unless it's polled ?

Reply:
mostly this, yes, and we also haven't added subscribe/unsubscribe methods.

Reply:
https://github.com/redis-rs/redis-rs/pull/1265